etl/sean.py

```diff
@@ -0,0 +1,68 @@
+'''
+Created on Dec 27, 2015
+@author: SeanEaster
+'''
+from math import floor
+
+
+import numpy as np
+import pandas as pd
+
+
+def standardize(array):
+    columnMeans = array.mean(axis=0)
+    return array - columnMeans
+
+# First, load the btc data and other assets
+btcDf = pd.DataFrame.from_csv('btc.csv')
+otherDf = pd.DataFrame.from_csv('data.csv')
+
+
+# running through different windows produces a lot of NaN values in rollingDeviations...this effs up np.divide later
+for _ in xrange(50):
+	STARTINDEX = np.random.randint(100)
+	LENGTH = np.random.randint(500)
+
+	btcValues = btcDf.ix[btcDf.index[STARTINDEX:(STARTINDEX + LENGTH)]].values.reshape((-1,))
+	meanBtcValues -= btcValues.mean()
+	btcStandardDev = meanBtcValues.std()
+
+	# Because numpy.correlate only supports single-dim arrays, we have to iterate
+	# Or I need to spend more time reading the docs. One of those two.
+
+	otherValues = otherDf.values
+	otherValues = standardize(otherValues)
+
+	rollingDeviations = pd.rolling_std(otherDf, LENGTH)#.values[(LENGTH - 1):, :]
```

Comment:
Ok, that helps a little, but I'm still very confused. Let's say I'm trying to match the period January 1, 2013 to January 1, 2014. The rolling window there is not 365 days long, because then...nothing is rolling, it's just the standard deviation of the time period. So, what is/should the window be, what is/should the stock price data set?